### ISSUE TITLE
Don't check metaglobal/keys/etc if /info/collections says they're unchanged

### ISSUE THREAD

**Issue by rfk** [mozilla/application-services#2617]
I instrumented a local Fenix build to log all network activity, and then mashed "sync now" a bunch of times to see what would happen. It looks like every sync will fetch `/info/collections`, then proceed to check `/meta/global/` and `/crypto/keys`:

```
02-11 14:53:44.096 17280 17325 W GeckoViewFetchClient: GV FETCH GET https://token.services.mozilla.com/1.0/sync/1.5
02-11 14:53:44.726 17280 17325 W GeckoViewFetchClient: GV FETCH RESPONSE GET https://token.services.mozilla.com/1.0/sync/1.5 200
02-11 14:53:44.726 17280 17325 W GeckoViewFetchClient: GV FETCH GET https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/info/configuration
02-11 14:53:45.506 17280 17325 W GeckoViewFetchClient: GV FETCH RESPONSE GET https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/info/configuration 200
02-11 14:53:45.516 17280 17325 W GeckoViewFetchClient: GV FETCH GET https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/info/collections
02-11 14:53:45.706 17280 17325 W GeckoViewFetchClient: GV FETCH RESPONSE GET https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/info/collections 200
02-11 14:53:45.716 17280 17325 W GeckoViewFetchClient: GV FETCH GET https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/storage/meta/global
02-11 14:53:45.916 17280 17325 W GeckoViewFetchClient: GV FETCH RESPONSE GET https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/storage/meta/global 200
02-11 14:53:45.916 17280 17325 W GeckoViewFetchClient: GV FETCH GET https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/storage/crypto/keys
02-11 14:53:46.126 17280 17325 W GeckoViewFetchClient: GV FETCH RESPONSE GET https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/storage/crypto/keys 200
[... more sync stuff here ...]
02-11 14:54:23.956 17280 17389 W GeckoViewFetchClient: GV FETCH GET https://token.services.mozilla.com/1.0/sync/1.5
02-11 14:54:24.526 17280 17389 W GeckoViewFetchClient: GV FETCH RESPONSE GET https://token.services.mozilla.com/1.0/sync/1.5 200
02-11 14:54:24.536 17280 17389 W GeckoViewFetchClient: GV FETCH GET https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/info/configuration
02-11 14:54:24.776 17280 17389 W GeckoViewFetchClient: GV FETCH RESPONSE GET https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/info/configuration 200
02-11 14:54:24.786 17280 17389 W GeckoViewFetchClient: GV FETCH GET https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/info/collections
02-11 14:54:25.016 17280 17389 W GeckoViewFetchClient: GV FETCH RESPONSE GET https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/info/collections 200
02-11 14:54:25.016 17280 17389 W GeckoViewFetchClient: GV FETCH GET https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/storage/meta/global
02-11 14:54:25.226 17280 17389 W GeckoViewFetchClient: GV FETCH RESPONSE GET https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/storage/meta/global 200
02-11 14:54:25.266 17280 17389 W GeckoViewFetchClient: GV FETCH GET https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/storage/crypto/keys
02-11 14:54:25.546 17280 17389 W GeckoViewFetchClient: GV FETCH RESPONSE GET https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/storage/crypto/keys 200
[... more sync stuff here ...]
```

I don't have any other devices syncing to this account, so I don't think there could have been any changes to these collections. ISTM that we should be able to look at the timestamps returned by `/info/collections`, see that `meta/global` and `crypto/keys` have not been modified since our last sync, and avoid the overhead of fetching and checking them.

For comparison, mashing "sync now" on desktop results in a fetch of `/info/collections` and then no further sync traffic, presumably because the reported timestamps all match the values it has already seen.

┆Issue is synchronized with this [Jira Bug](https://jira.mozilla.com/browse/SYNC-812)
┆Story Points: 2
┆Epic: Fenix Migration
┆Sprint: SYNC - end 2020-02-28

**Comment by data-sync-user** [comment]
➤ Janet Dragojevic commented:

Take 1/2 day to determine if it's a relatively simple fix.

**Comment by mhammond** [comment]
Also fixed by #2664

**Comment by rfk** [comment]
> Also fixed by #2664

Unfortunately, I don't think it was. Running the latest master of appservices I see the following on each "sync now":

```
43795:02-24 14:47:31.371 19718 19739 W GeckoViewFetchClient: FETCH GET https://token.services.mozilla.com/1.0/sync/1.5)
43816:02-24 14:47:32.051 19718 19739 W GeckoViewFetchClient: FETCH GET https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/info/configuration)
43868:02-24 14:47:32.641 19718 19739 W GeckoViewFetchClient: FETCH GET https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/info/collections)
43891:02-24 14:47:32.841 19718 19739 W GeckoViewFetchClient: FETCH GET https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/storage/meta/global)
43905:02-24 14:47:33.041 19718 19739 W GeckoViewFetchClient: FETCH GET https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/storage/crypto/keys)
43912:02-24 14:47:33.251 19718 19739 W GeckoViewFetchClient: FETCH GET https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/storage/clients?full=1)
43918:02-24 14:47:33.451 19718 19739 W GeckoViewFetchClient: FETCH POST https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/storage/clients?batch=true&commit=true)

44049:02-24 14:47:33.891 19718 19740 W GeckoViewFetchClient: FETCH GET https://token.services.mozilla.com/1.0/sync/1.5)
44068:02-24 14:47:34.121 19718 19740 W GeckoViewFetchClient: FETCH GET https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/info/configuration)
44077:02-24 14:47:34.311 19718 19740 W GeckoViewFetchClient: FETCH GET https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/info/collections)
44081:02-24 14:47:34.511 19718 19740 W GeckoViewFetchClient: FETCH GET https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/storage/meta/global)
44094:02-24 14:47:34.751 19718 19740 W GeckoViewFetchClient: FETCH GET https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/storage/crypto/keys)
44104:02-24 14:47:34.991 19718 19740 W GeckoViewFetchClient: FETCH GET https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/storage/clients?full=1)
44109:02-24 14:47:35.231 19718 19740 W GeckoViewFetchClient: FETCH POST https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/storage/clients?batch=true&commit=true)

44940:02-24 14:48:01.141 19718 19739 W GeckoViewFetchClient: FETCH GET https://token.services.mozilla.com/1.0/sync/1.5)
44948:02-24 14:48:01.861 19718 19739 W GeckoViewFetchClient: FETCH GET https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/info/configuration)
44952:02-24 14:48:02.091 19718 19739 W GeckoViewFetchClient: FETCH GET https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/info/collections)
44956:02-24 14:48:02.321 19718 19739 W GeckoViewFetchClient: FETCH GET https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/storage/meta/global)
44964:02-24 14:48:02.601 19718 19739 W GeckoViewFetchClient: FETCH GET https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/storage/crypto/keys)
44971:02-24 14:48:02.841 19718 19739 W GeckoViewFetchClient: FETCH GET https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/storage/clients?full=1)
44976:02-24 14:48:03.061 19718 19739 W GeckoViewFetchClient: FETCH POST https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/storage/clients?batch=true&commit=true)

45643:02-24 14:48:36.951 19718 19827 W GeckoViewFetchClient: FETCH GET https://token.services.mozilla.com/1.0/sync/1.5)
45652:02-24 14:48:37.611 19718 19827 W GeckoViewFetchClient: FETCH GET https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/info/configuration)
45656:02-24 14:48:37.841 19718 19827 W GeckoViewFetchClient: FETCH GET https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/info/collections)
45660:02-24 14:48:38.131 19718 19827 W GeckoViewFetchClient: FETCH GET https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/storage/meta/global)
45668:02-24 14:48:38.421 19718 19827 W GeckoViewFetchClient: FETCH GET https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/storage/crypto/keys)
45675:02-24 14:48:38.741 19718 19827 W GeckoViewFetchClient: FETCH GET https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/storage/clients?full=1)
45680:02-24 14:48:39.061 19718 19827 W GeckoViewFetchClient: FETCH POST https://sync-743-us-west-2.sync.services.mozilla.com/1.5/130981380/storage/clients?batch=true&commit=true)
```

#2664 does seem to have prevented unnecessary calls to the individual item collections like `bookmarks` and `history`, but not to the meta-level collections such as `meta` and `keys`.

**Comment by mhammond** [comment]
Fixed by #2711
